### PR TITLE
Add Flux function `union` to SPEC

### PR DESCRIPTION
**Pull request by wolffcm**
I treated `join()` as a template where applicable, and included an example so semantics are clear.

Question: should the parameter name be called `streams` instead of `tables`?  This question applies to `join()` too.

### CODE REVIEW

File: docs/SPEC.md [hunk 1]
```diff
@@ -2146,6 +2146,69 @@ Examples:
...
+
+##### Options
+
+| Name   | Type | Required | Default Value | Possible Values |
+| ------ | ---- | -------- | ------------- | --------------- |
+| tables | map  | yes      |  none         |  N/A            |
+
+##### Output Schema
+
+The output schema of union shall be identical to the input schema.
```
Comment (aanthony1243) [inline docs/SPEC.md]: ...shall be the union of all input schemas.

File: docs/SPEC.md [hunk 2]
```diff
@@ -2146,6 +2146,69 @@ Examples:
...
+    | ----- | ------ | ------ |
+    | 0001  | "temp" | 55 |
+    | 0002  | "temp" | 56 |
+
+    | _time | _field | _value |
+    | ----- | ------ | ------ |
+    | 0001  | "pressure" | 29.82 |
+    | 0002  | "pressure" | 30.01 |
+
+`union(tables: {sf: SF_Temperature, ny: NY_Temperature})` produces:
```
Comment (aanthony1243) [inline docs/SPEC.md]: @jlapacik   what happens if there's > 2 tables for join?   

@wolffcm  do we want to support N >= 2 tables for union?  Seems like it wouldn't be much more complex than the N=2 case.

Comment (jlapacik) [reply docs/SPEC.md]: @aanthony1243 join does not support joining more that two streams at the moment.

Comment (wolffcm) [reply docs/SPEC.md]: It seems like we could handle more than 2 streams with no change at all to the processing algorithm.  Is there any reason not to support an arbitrary number of of tables?

Comment (jlapacik) [reply docs/SPEC.md]: I don't think so. I would say do it if it's trivial.

Comment (aanthony1243) [reply docs/SPEC.md]: @jlapacik probably not urgent, but the spec leaves it undefined what happens if you pass >2 tables to join

File: docs/SPEC.md [hunk 3]
```diff
@@ -2146,6 +2146,69 @@ Examples:
...
+    | ----- | ------ | ------ |
+    | 0001  | "humidity" | 81 |
+    | 0002  | "humidity" | 82 |
+
+    | _time | _field | _value |
+    | ----- | ------ | ------ |
+    | 0001  | "pressure" | 29.82 |
+    | 0002  | "pressure" | 30.01 |
+
+##### Options
```
Comment (jlapacik) [inline docs/SPEC.md]: Can you remove this options sub-section and replace it with: "Union has the following properties: ... " under the initial description. All specs should follow this convention but Join does not.

File: docs/SPEC.md [hunk 4]
```diff
@@ -2146,6 +2146,69 @@ Examples:
     | 0002  | "temp" | 75 | "temp" | 56 |
     | 0003  | "temp" | 72 | "temp: | 55 |
 
+#### Union
+
+Union concatenates two input streams into a single output stream.  In tables that have identical
+schema and group keys, contents of the tables will be concatenated in the output stream.
+
```
Comment (jlapacik) [inline docs/SPEC.md]: Can you list the properties (or in this case property) of Union under this description?

File: docs/SPEC.md (hunk 2 above)
Comment (jlapacik) [reply docs/SPEC.md]: @aanthony1243 there are also a few other things I need to change in the join spec. I'll put up a follow-up PR.

Comment (nathanielc) [reply docs/SPEC.md]: >  Should it use a simple list of streams instead?

I was thinking that union should just take a list of tables.

**Comment by nathanielc** [comment]
> Question: should the parameter name be called streams instead of tables? This question applies to join() too.

The parameter name should be `tables`.

**Comment by jlapacik** [comment]
@wolffcm The example tables don't seem to be getting rendered correctly. Perhaps remove the tabs?

**Comment by wolffcm** [comment]
@jlapacik Okay, I fixed the markdown so the tables render.  I had to indent with only three spaces for some reason, even though other tables indent with 4 and render just fine.

**Pull request merged**